swing/src/net/sf/openrocket/gui/main/SimulationPanel.java

```diff
@@ -598,6 +624,216 @@ public void updateEnabledState() {
 		}
 	}
 
+	class DumpSimulationToCSVAction extends SimulationAction {
```

Comment:
I don't really like that the code for the actual CSV writing is implemented here, doesn't make the code very modular. Instead, take a look at `core/src/net/sf/openrocket/file/CSVExport.java`. I would create a separate file in `core/src/net/sf/openrocket/file` called `SimulationTableCSVExport.java` and implement the CSV exporting functionality there. Or if you find out that `CSVExport.java` contains a lot of the same code that you implemented here, it may be better to put your code in `CSVExport.java` and then use/rewrite some of the existing code to fit your goal. (that was one of my first concerns when I watched your code, that there may be a lot of duplicate code with the code in `CSVExport.java`. I haven't dug in the code to verify that, my apologies, but be careful with this.)

Reply:
This will take some time...

Reply:
Time is all we have :) By the way, we appreciate all your hard work on this 👍 All these comments sound like nagging, so I hope you're not offended by them. I know what it feels like when you make your first contributions to an open-source project and you then get a thousand comments from seniors. It's not fun, but it improves the program 🙂.

Reply:
Did the initial refactor but when I tried to move it under core had some issues... so for now it's still in openrocket-swing

Reply:
You need to make sure that when you refactor the file to Core that there aren't any references to Java classes from Java Swing, otherwise you'll get cyclic dependency issues. So just refactor your code well enough so that all the info that the CSV export needs from the Swing-part of OR, it gets from method arguments.

Reply:
But if there's no easy way to do that, it's perfectly fine to put it in `swing/src/net/sf/openrocket/file`.